### PR TITLE
Tolerate head that is None in get_attributes

### PR DESCRIPTION
**Pull request by rocky**
Fixes issue #203

Also remove some duplicate imports

(After merge I'll update CHNAGES.rst - to do so now will just cause merge conflicts.)

<a href="https://gitpod.io/#https://github.com/Mathics3/mathics-core/pull/208"></a>

### CODE REVIEW

**Review by TiagoCavalcante — APPROVED** [review]
LGTM

File: mathics/builtin/base.py [hunk 1]
```diff
@@ -859,6 +860,8 @@ def get_match_candidates(self, leaves, expression, attributes, evaluation, vars=
         return leaves
 
     def get_attributes(self, definitions):
+        if self.head is None:
```
Comment (mmatera) [inline mathics/builtin/base.py:863]: That is weird: why the head of a pattern would be None?

Comment (rocky) [reply mathics/builtin/base.py:863]: https://github.com/Mathics3/mathics-core/blob/master/mathics/builtin/patterns.py#L1158 which goes back to th every first versions.

Comment (TiagoCavalcante) [reply mathics/builtin/base.py:863]: @rocky I believe this deserves a comment.

Comment (rocky) [reply mathics/builtin/base.py:863]: @TiagoCavalcante feel free to add one.

Comment (TiagoCavalcante) [reply mathics/builtin/base.py:864]: Done.

Comment (TiagoCavalcante) [reply mathics/builtin/base.py:864]: Done.

File: mathics/builtin/base.py [hunk 2]
```diff
@@ -844,25 +844,33 @@ def error(self, tag, *args):
...
-        return self.get_name()
+    def get_attributes(self, definitions) -> int:
+        """
+        If self has head return head's attributes.
+        Otherwise that no attributes have been set.
+        """
+        if self.head is None:
+            # FIXME: _Blank in builtin/patterns.py sets head to None.
+            # Figure out if this is the best thing to do and explain why.
+            return nothing
```
Comment (rocky) [inline mathics/builtin/base.py:855]: @TiagoCavalcante it really bothers me what we have an attribute called `nothing`.  Please consider changing it to `no_attributes`. Thanks.

Comment (TiagoCavalcante) [reply mathics/builtin/base.py:855]: no_attributes seems better, I'll change it after merging this PR.

**Review by mmatera — APPROVED** [review]
LGTM. Afterward, we can see what is going wrong in create_pattern.

**Pull request merged**